### PR TITLE
Add black to requirements and pre-commit 

### PR DESCRIPTION
**Pull request by caneff**

### CODE REVIEW

**Comment by caneff** [comment]
OK I think I addressed all your points from the other PR. Sorry it was easier for me to just close that one and continue here.

File: docs/source/deepdive_into_dtypes.ipynb [hunk 1]
```diff
@@ -142,11 +145,9 @@
     "    a: np.integer\n",
     "    b: np.float_\n",
     "\n",
+    "\n",
     "df = DataSet[Schema](\n",
-    "    {\n",
-    "        \"a\": pd.Series([1, 2, 3], dtype=np.int64),\n",
-    "        \"b\": pd.Series([1.0, 2.0, 3.0], dtype=np.float64)\n",
-    "    }\n",
+    "    {\"a\": pd.Series([1, 2, 3], dtype=np.int64), \"b\": pd.Series([1.0, 2.0, 3.0], dtype=np.float64)}\n",
```
Comment (nanne-aben) [inline docs/source/deepdive_into_dtypes.ipynb:150]:
```
{"a": pd.Series([1, 2, 3], dtype=np.int64), "b": pd.Series([1.0, 2.0, 3.0], dtype=np.float64),}"
```

File: docs/source/deepdive_into_dtypes.ipynb [hunk 2]
```diff
@@ -224,14 +227,13 @@
...
     "class SchemaB:\n",
     "    a: np.int64\n",
     "\n",
+    "\n",
     "try:\n",
-    "    (\n",
-    "        DataSet[SchemaA]()\n",
-    "        .pipe(DataSet[SchemaB])\n",
-    "    )\n",
+    "    (DataSet[SchemaA]().pipe(DataSet[SchemaB]))\n",
```
Comment (nanne-aben) [inline docs/source/deepdive_into_dtypes.ipynb:236]:
```
DataSet[SchemaA]().pipe(DataSet[SchemaB])
```

File: docs/source/deepdive_into_dtypes.ipynb [hunk 3]
```diff
@@ -382,12 +376,8 @@
...
     "    b: Any\n",
     "\n",
-    "df = DataSet[Schema](\n",
-    "    {\n",
-    "        \"a\": [1, 2, 3],\n",
-    "        \"b\": [\"1\", 2, None]\n",
-    "    }\n",
-    ")\n",
+    "\n",
+    "df = DataSet[Schema]({\"a\": [1, 2, 3], \"b\": [\"1\", 2, None]})\n",
```
Comment (nanne-aben) [inline docs/source/deepdive_into_dtypes.ipynb:380]:
```
df = DataSet[Schema]({"a": [1, 2, 3], "b": [\"1\", 2, None],})
```

File: docs/source/getting_started.ipynb [hunk 4]
```diff
@@ -187,13 +192,10 @@
...
     "\n",
+    "\n",
     "def foo() -> DataSet[Schema]:\n",
     "    return DataSet[Schema](\n",
-    "        {\n",
-    "            \"id\": [1, 2, 3],\n",
-    "            \"name\": [\"John\", \"Jane\", \"Jack\"],\n",
-    "            \"job\": \"Data Scientist\"\n",
-    "        }\n",
+    "        {\"id\": [1, 2, 3], \"name\": [\"John\", \"Jane\", \"Jack\"], \"job\": \"Data Scientist\"}\n",
```
Comment (nanne-aben) [inline docs/source/getting_started.ipynb:198]:
```
{"id": [1, 2, 3], "name": ["John", "Jane", "Jack"], "job": "Data Scientist",}
```

File: docs/source/getting_started.ipynb [hunk 5]
```diff
@@ -356,10 +361,7 @@
    "outputs": [],
    "source": [
     "def foo(data: DataSet[SchemaA]) -> DataSet[SchemaB]:\n",
-    "    return (\n",
-    "        df.assign(id=lambda df: range(df.shape[0]))\n",
-    "        .pipe(DataSet[SchemaB])\n",
-    "    )"
+    "    return df.assign(id=lambda df: range(df.shape[0])).pipe(DataSet[SchemaB])"
```
Comment (nanne-aben) [inline docs/source/getting_started.ipynb:364]:
```
return df.assign(id=lambda df: range(df.shape[0]),).pipe(DataSet[SchemaB])
```

**Review by nanne-aben — COMMENTED** [review]
Sorry to be nitpicking, but I feel like the code in the notebooks is sometimes formatted a bit funky. I've made some suggestions to add a `,` in some places. If I'm not mistaken, black will use these to reformat these pieces of codes with some newlines, making them more readable. I've also made suggestions to remove some redundant `()`.

**Comment by caneff** [comment]
OK think I did them all.

**Review by nanne-aben — APPROVED** [review]
Looks good, thanks for your contribution!

I pushed one commit myself, wanted to make some more small edits in the formatting of the notebooks and didn't want to bother you with that :)

**Pull request merged**